### PR TITLE
LPS-120432 Remove unnecessary useMemo occurrences and improve the metrics pages performance

### PR DESCRIPTION
**Pull request by albinojunior**

### CODE REVIEW

**Comment by albinojunior** [comment]
ci:test:sf

**Comment by albinojunior** [comment]
ci:test:relevant

**Comment by rafaprax** [comment]
ci:test:relevant

**Comment by albinojunior** [comment]
ci:test:workflow

**Comment by albinojunior** [comment]
ci:test:sf

**Comment by albinojunior** [comment]
ci:test:relevant

**Comment by albinojunior** [comment]
ci:test:workflow

**Comment by albinojunior** [comment]
ci:test:sf

**Comment by albinojunior** [comment]
ci:test:relevant

**Comment by albinojunior** [comment]
ci:test:workflow

**Comment by albinojunior** [comment]
ci:test:relevant

**Comment by albinojunior** [comment]
ci:test:sf

**Comment by albinojunior** [comment]
ci:test:workflow

**Comment by albinojunior** [comment]
ci:test:sf

**Comment by albinojunior** [comment]
ci:test:relevant

**Comment by rafaprax** [comment]
ci:help

**Comment by albinojunior** [comment]
@rafaprax i will forward this PR. Is it ok for you?

**Comment by albinojunior** [comment]
ci:forward

**Comment by rafaprax** [comment]
Just started reviewing :)

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/InstanceListPageHeader.es.js [hunk 1]
```diff
@@ -154,28 +145,12 @@ const Header = ({
...
-	const {key} = statusesFilterItem ? statusesFilterItem.items[0] : {};
-	const completedStatusSelected = useMemo(
-		() =>
-			selectedFilters.length > 0 && statusesFilterItem
-				? key === processStatusConstants.completed
-				: false,
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-		[key]
-	);
+	const isCompletedSelected = selectedFilters
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/InstanceListPageHeader.es.js:148]: @albinojunior Rename to `isCompletedStatusSelected `, please

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/reassign/bulk/BulkReassignModal.es.js [hunk 2]
```diff
@@ -136,71 +136,52 @@ const BulkReassignModal = () => {
...
-					order: 1,
-					previousBtn: false,
-					props: {setErrorToast},
-					subtitle: Liferay.Language.get('select-tasks'),
-					title: Liferay.Language.get('select-tasks-to-reassign'),
-				},
-			};
-
-			return steps[step];
+	const STEPS = {
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/reassign/bulk/BulkReassignModal.es.js:139]: @albinojunior , use lowercase instead, please

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/InstanceListPageHeader.es.js [hunk 3]
```diff
@@ -154,28 +145,12 @@ const Header = ({
...
+		.find(({key}) => key === filterConstants.processStatus.key)
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/InstanceListPageHeader.es.js:149]: @albinojunior Rename to `completedStatusSelected `, please

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/transition/bulk/BulkTransitionModal.es.js [hunk 4]
```diff
@@ -141,68 +141,51 @@ const BulkTransitionModal = () => {
...
-					},
-					props: {setErrorToast},
-					subtitle: Liferay.Language.get('choose-transition'),
-					title: Liferay.Language.get('choose-transition-per-step'),
-				},
-			};
-
-			return steps[step];
+	const STEPS = {
+		selectTasks: {
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/transition/bulk/BulkTransitionModal.es.js:145]: @albinojunior , use lowercase instead, please

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/update-due-date/BulkUpdateDueDateModal.es.js [hunk 5]
```diff
@@ -136,71 +132,53 @@ const BulkUpdateDueDateModal = () => {
...
-					previousBtn: false,
-					props: {setErrorToast},
-					subtitle: Liferay.Language.get('select-tasks'),
-					title: Liferay.Language.get('select-tasks-to-update'),
-				},
-			};
-
-			return steps[step];
+	const STEPS = {
+		selectDueDate: {
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/update-due-date/BulkUpdateDueDateModal.es.js:136]: @albinojunior , use lowercase instead, please

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/update-due-date/UpdateDueDateStep.es.js [hunk 6]
```diff
@@ -152,13 +152,10 @@ const TimePickerInputWithOptions = ({format, isAmPm, setValue, value}) => {
...
-	const options = useMemo(() => getTimeOptions(isAmPm), [isAmPm]);
-	const popoverStyle = useMemo(() => {
-		const {current: {offsetWidth = 270} = {}} = inputRef;
+	const options = getTimeOptions(isAmPm);
 
-		return {left: `${(offsetWidth - 120) / 2}px`};
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [inputRef.current]);
+	const {offsetWidth = 270} = inputRef.current || {};
+	const popoverStyle = {left: `${(offsetWidth - 120) / 2}px`};
```
Comment (rafaprax) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/update-due-date/UpdateDueDateStep.es.js:158]: @albinojunior, add linebreak betwwen statements, please

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Review by rafaprax — APPROVED** [review]

**Comment by rafaprax** [comment]
ci:test:workflow

**Comment by rafaprax** [comment]
ci:test:relevant

**Comment by albinojunior** [comment]
ci:test:relevant

**Comment by albinojunior** [comment]
ci:test:sf

**Comment by albinojunior** [comment]
ci:test:workflow

**Comment by rafaprax** [comment]
ci:test:relevant

**Comment by rafaprax** [comment]
ci:forward

**Pull request closed without merging**